wdl-lint/src/rules/nonmatching_output.rs

```diff
@@ -0,0 +1,249 @@
+                )
+            })
+            .collect();
+
+        let mut extra_found = false;
+
+        // Check for entries missing from `meta`.
+        for (name, span) in &expected {
+            if !actual.contains_key(name) {
+                if !extra_found {
+                    extra_found = true;
+                }
+                state.add(nonmatching_output(*span, name, &context));
+            }
+        }
+
+        // Check for extra entries in `meta`.
+        for (name, span) in &actual {
+            if !expected.contains_key(name) {
+                if !extra_found {
+                    extra_found = true;
+                }
+                state.add(extra_output_in_meta(*span, name, &context));
+            }
+        }
+
+        // Check for out-of-order entries.
+        if !extra_found && !actual.keys().eq(expected.keys()) {
+            state.add(out_of_order(
+                meta_outputs_key.syntax().text_range().to_span(),
+                outputs.syntax().text_range().to_span(),
+                &context,
+            ));
+        }
+    } else {
+        state.add(missing_outputs_in_meta(
+            meta.syntax().first_token().unwrap().text_range().to_span(),
+            &context,
+        ));
+    }
+}
+
+impl Visitor for NonmatchingOutputRule {
+    type State = Diagnostics;
+
+    fn document(&mut self, _: &mut Self::State, reason: VisitReason, _: &Document) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        // Reset the visitor upon document entry
+        *self = Default::default();
+    }
+
+    fn workflow_definition(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        workflow: &WorkflowDefinition,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        // If the `meta` section is missing, the MissingMeta rule will handle it.
+        if let Some(meta) = workflow.metadata().next() {
+            // If the `output` section is missing, the MissingOutput rule will handle it.
+            if let Some(output) = workflow.outputs().next() {
+                if output.declarations().count() > 0 {
+                    check_output_meta(
+                        state,
+                        &meta,
+                        &output,
+                        TaskOrWorkflow::Workflow(workflow.clone()),
+                    );
+                }
+            }
+        }
+    }
+
+    fn task_definition(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        task: &TaskDefinition,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        // If the `meta` section is missing, the MissingMeta rule will handle it.
+        if let Some(meta) = task.metadata().next() {
+            // If the `output` section is missing, the MissingOutput rule will handle it.
+            if let Some(output) = task.outputs().next() {
```

Comment:
```suggestion
            if let Some(outputs) = task.outputs().next() {
```

to match the variable name in `check_output_meta`.